backend/src/main/java/com/htecgroup/skynest/filter/CustomAuthenticationFilter.java

```diff
@@ -80,15 +79,14 @@ protected void successfulAuthentication(
       Authentication authentication)
       throws IOException, ServletException {
 
-    User user = (User) authentication.getPrincipal();
+    LoggedUserDto user = (LoggedUserDto) authentication.getPrincipal();
+    List<String> authorities = user.getRoleNames();
 
-    List<String> authorities =
-        user.getAuthorities().stream()
-            .map(GrantedAuthority::getAuthority)
-            .collect(Collectors.toList());
-    String token =
-        JwtUtils.generate(
-            user.getUsername(), JwtUtils.ACCESS_TOKEN_EXPIRATION_MS, "roles", authorities);
+    JwtObject jwtObject = new JwtObject();
```

Comment:
You can create a all args constructor and set those props there instead of setter here. having that said, setter is not needed